packages/datadog-plugin-aws-sdk/src/helpers.js

```diff
@@ -0,0 +1,172 @@
+'use strict'
+
+const helpers = {
+  finish (span, err) {
+    if (err) {
+      span.setTag('error', err)
+      this.addRequestIdTag(span, err)
+    }
+
+    span.finish()
+  },
+
+  addResponseTags (span, response, serviceName, config) {
+    if (!span) return
+
+    if (response.request) {
+      this.addServicesTags(span, response, serviceName)
+    }
+
+    this.addRequestIdTag(span, response)
+    config.hooks.request(span, response)
+  },
+
+  addServicesTags (span, response, serviceName) {
+    if (!span) return
+
+    const tags = {}
+    const params = response.request.params
+    const operation = response.request.operation
+
+    if (operation && params) {
+      switch (serviceName) {
+        case 'dynamodb':
+          this.addDynamoDbTags(params, operation, tags)
+          break
+
+        case 'kinesis':
+          this.addKinesisTags(params, operation, tags)
+          break
+
+        case 's3':
+          this.addS3Tags(params, operation, tags)
+          break
+
+        case 'sqs':
+          this.addSqsTags(params, operation, tags)
+          break
+
+        case 'sns':
+          this.addSnsTags(params, operation, response, tags)
+          break
+
+        case 'cloudwatchlogs':
+          this.addCwLogsTags(params, operation, tags)
+          break
+      }
+    }
+
+    this.addDefaultTags(operation, tags)
+    span.addTags(tags)
+  },
+
+  addDefaultTags (operation, tags) {
+    // defaults
+    if (tags['resource.name']) return
+
+    Object.assign(tags, {
+      'resource.name': operation || 'Amazon'
```

Comment:
I'd leave it blank by default, meaning remove the fallback, which will make it use `aws.request`.

Reply:
fair enough, removed